yjit/src/backend/arm64/mod.rs

```diff
@@ -961,6 +961,16 @@ impl Assembler
             };
         }
 
+        // Invalidate icache for newly written out region so we don't run
+        // stale code.
+        {
+            let start = cb.get_ptr(start_write_pos).raw_ptr();
+            let write_ptr = cb.get_write_ptr().raw_ptr();
+            let codeblock_end = cb.get_ptr(cb.get_mem_size()).raw_ptr();
+            let end = std::cmp::min(write_ptr, codeblock_end);
+            unsafe { rb_yjit_icache_invalidate(start as _, end as _) };
+        }
```

Comment:
Thanks for putting this together Alan!

Do you know if `__builtin___clear_cache(start, end);` uses a syscall underneath, or just a machine loop? Asking because if it uses a syscall, we might be able to batch multiple calls together in a future PR to improve speed further.